Feature/81 embedded tauri

pospi opened the pull request:
This is the start of an integration, but I'm aborting before I get sucked further into whatever Nix' equivalent of dependency hell is.

`src-tauri` will currently not compile since the Rust toolchain bundled with Holochain's base Nix flake is a `1.66.1` `rustc` and Tauri requires `1.70.0` or newer.

The base config appears to be inherited from https://github.com/oxalica/rust-overlay but I am not seeing any version pinning at Holochain's layer- potentially this is the latest version of the compiler provided in that stack. I'm unsure whether this means the entire toolchain has to be reconfigured or whether it is possible to override / "side-load" a newer `rustc` into the environment.

9ed8070 was a test to see if pinning to the latest Holochain version (which we should be doing anyway) would get us a newer compiler. It did not. So you may find this useful as a basis for more changes or prefer to rebase it out and change the entire `flake.nix`.

Sub-items required to complete this task:

1. [ ] Get the Nix configuration for Mac working and ensure the Linux configuration functions as expected (see TODO in `flake.nix`)
2. [ ] Fix rendering crash on Linux. This appears to be related to some missing dependencies:
```
MESA-LOADER: failed to open iris: /run/opengl-driver/lib/dri/iris_dri.so: cannot open shared object file: No such file or directory (search paths /run/opengl-driver/lib/dri, suffix _dri)
MESA-LOADER: failed to open zink: /run/opengl-driver/lib/dri/zink_dri.so: cannot open shared object file: No such file or directory (search paths /run/opengl-driver/lib/dri, suffix _dri)
MESA-LOADER: failed to open kms_swrast: /run/opengl-driver/lib/dri/kms_swrast_dri.so: cannot open shared object file: No such file or directory (search paths /run/opengl-driver/lib/dri, suffix _dri)
MESA-LOADER: failed to open swrast: /run/opengl-driver/lib/dri/swrast_dri.so: cannot open shared object file: No such file or directory (search paths /run/opengl-driver/lib/dri, suffix _dri)
Failed to create GBM device for render device: /dev/dri/renderD128: No such file or directory
Cannot get default EGL display: EGL_BAD_PARAMETER
Cannot create EGL context: invalid display (last error: EGL_SUCCESS)
```
3. [ ] Ensure DevTools panel renders correctly on Linux (it looks like Stevee J has [already put some work into this in Holochain core](https://github.com/tauri-apps/tauri/issues/5711#issuecomment-1482705393))

These three issues may all be related- at the moment I can't get to the DevTools check because triggering the panel causes the same failure as per point 2 above.

pospi committed:
stock Tauri app output from project initiator

pospi committed:
add Tauri app runtime crate to workspace

pospi committed:
update lockfiles

pospi committed:
pin Holochain version to current latest release. May cause compilation bugs.

adaburrows commented:
It looks like the hc-launch pinned an old version, tauri-build v1.4 instead of v1.5 or v2.0: https://github.com/holochain/launcher/blob/develop/crates/hc_launch/src-tauri/Cargo.toml#L20C31-L20C31

pospi committed:
Merge branch 'develop' into feature/81-embedded-tauri

pospi commented:
I still had linker errors on `cfg-expr`, but the [latest commits](https://github.com/holochain/holochain/pull/2660) into the Holochain Nix env bump `rustc` to 1.77.1. I am testing that now.

It's likely we will need to do a project-wide sweep for `hdi`/`hdk` versions and run through compatibility updates to Holochain fairly soon.

pospi committed:
pin to latest Holonix ref to get rustc 1.71.1

pospi committed:
bump tauri versions for updated Rust toolchain

pospi committed:
separate Tauri build from workspace for compilation to non-WASM target & fix Tauri build deps for Linux

pospi committed:
update lockfiles

pospi commented:
Made good progress- our outdated toolchain seems to still _compile_ fine with this Holonix env. Remains to be seen whether it _executes_, but I need to do a pass at the package scripts and wire them into Tauri better in order to test that.

pospi committed:
update package scripts for single-agent execution via Tauri

multi-agent scripts should be implementable from here, by concurrently repeating 'start' with different env vars

also, dotenv needs removing

pospi committed:
update Tauri CLI dependency

pospi committed:
set Applet system identifier

pospi committed:
update lockfile
